docs/features/plugin/tools/getting-started.md

```diff
@@ -0,0 +1,217 @@
+---
+sidebar_position: 5
+title: "🔧📖  Tool Development Guide"
+---
+# 🔧📖 Tool Development Guide
+This documentation provides a complete guide for developers to create tools compatible with Open WebUI. By following these guidelines, you can ensure that your tools are efficient, user-friendly, and integrate well with the platform.
+
+> **Tip:**  A great way to get more familiar with tools is by exploring [community tools](https://openwebui.com/tools/).
+---
+
+## Introduction
+Open WebUI allows developers to extend its functionality by building custom tools using **pure python**. To maintain consistency and usability across the platform, it is essential to follow the development standards outlined in this guide.
+
+---
+
+## Class Structure
+All tools must be encapsulated in a class named `Tools`. This ensures compatibility with Open WebUI’s architecture.
+
+```python 
+class Tools:
+    def __init__(self):
+        pass
+```
+### file_handler:
+To define a file handler, you need to set the file_handler attribute in your tool's module. This attribute indicates whether the tool has custom logic for handling files. Using file handler in your tool will prevent open webui to later process the files.
+Here's an example:
+
+```python
+class Tools:
+    def __init__(self):
+        self.file_handler = True  # Indicates custom file handling logic
+    def process_file(self, file_path):
+        # Custom file processing logic
+        pass
+```
+
+### citation:
+To share the citation to the sources used by the tool you can use the citation keyword inside your tool. Please check the [event emitters](#event-emitters) section for more information.
+Here is an example:
+
+```python
+class Tools:
+    def __init__(self):
+        self.citation = True
+```
+<!-- --- -->
+<!-- ## Import Path Replacements
+You can integrate Open WebUI functionalities into your code by using its standardized import paths. These paths allow you to organize and access tools efficiently, supporting streamlined development. The available import paths include from utils, from main, from config, and from apps.
+
+```python
+from config import ADMIN_EMAIL
+``` -->
+<!-- --- -->
+## Frontmatter Requirements
+Each tool should define its requirements and parameters in the frontmatter. The frontmatter must:
+
+1. Begin and end with triple backticks (""").
+
+2. Start with a description (optional), followed by key-value pairs for parameters.
+
+### Requirements: 
+A list of Python packages that the tool depends on. These are specified as a comma-separated string and are installed using pip. Example:
```

Comment:
Mention it's automatically downloaded if specified in the frontmatter (no need to specify if  one of open webui dependencies).